libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -1116,26 +1116,25 @@ class MapboxNavigation @VisibleForTesting internal constructor(
      * By default uses [MapboxRerouteController].
      */
     fun setRerouteController(rerouteController: RerouteController) {
-        val currentController = this.rerouteController
-        this.rerouteController = LegacyRerouteControllerAdapter(rerouteController)
-
-        if (currentController?.state == RerouteState.FetchingRoute) {
-            currentController.interrupt()
-            reroute()
-        }
+        setRerouteController(
+            LegacyRerouteControllerAdapter(rerouteController) as NavigationRerouteController?
+        )
     }
 
     /**
-     * Set [RerouteOptionsAdapter]. It allows to modify [RouteOptions] before a reroute request
-     * is sent if the default reroute controller is used. Pass `null` to clear the adapter.
+     * Set [NavigationRerouteController] that's automatically invoked when user is off-route.
+     *
+     * By default [MapboxRerouteController] is used.
+     * Pass `null` to clear the adapter.
```

Comment:
`Adapter` is a legacy name
```suggestion
     * Pass `null` to disable the controller.
```